src/Microsoft.ML.AutoML/TransformInference/TransformInference.cs

```diff
@@ -210,6 +214,29 @@ public override IEnumerable<SuggestedTransform> Apply(IntermediateColumn[] colum
                 }
             }
 
+            internal sealed class GroupId : TransformInferenceExpertBase
+            {
+                public GroupId(MLContext context) : base(context)
+                {
+                }
+
+                public override IEnumerable<SuggestedTransform> Apply(IntermediateColumn[] columns, TaskKind task)
+                {
+                    if (task != TaskKind.Ranking)
+                    {
+                        yield break;
+                    }
+                    foreach (var column in columns)
+                    {
+                        if (column.Purpose == ColumnPurpose.GroupId && !column.Type.IsKey())
```

Comment:
Does that has to use OneHotHashEncoding?

Maybe reuse Catagorical experts?

Reply:
Yes, I talked to @justinormont  and the reason for using OneHotHashEncoding is 1) to better distribute data across different folds (since larger groupIds tend to appear earlier in a dataset), 2) hashing is faster 3) scoring for unseen groupIds.